src/Workspaces/Core/MSBuild.BuildHost/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -386,132 +393,75 @@ public void RemoveDocument(string filePath)
             }
         }
 
-        public void AddMetadataReference(MetadataReference reference, AssemblyIdentity identity)
+        public void AddMetadataReference(string metadataReferenceIdentity, MetadataReferenceProperties properties, string? hintPath)
```

Comment:
The rest of this file are MSBuildWorkspace's support for TryApplyChanges. Previously this would be called in the same process of the Roslyn workspace, so it could pass around workspace objects directly. Now this is on the server side of the RPC channel, so we need simpler types. You'll see some of this logic is now in MSBuildWorkspace directly.